Library/Homebrew/dev-cmd/pull.rb

```diff
@@ -287,7 +287,8 @@ def merge_commit?(url)
     pr_number = url[%r{/pull\/([0-9]+)}, 1]
     return false unless pr_number
 
-    safe_system "git", "fetch", "--quiet", "origin", "pull/#{pr_number}/head"
+    remote = ENV["HOMEBREW_FORCE_HOMEBREW_ON_LINUX"] ? "homebrew" : "origin"
```

Comment:
Could we get a comment explaining why this is doing what it is?

Reply:
Done, I think.

Reply:
Sorry to be pedantic but that explains what rather than why. What I'm interested in is e.g. where does the `homebrew` git remote name come from?

Reply:
Changed.

Reply:
There's also further context in the first few paragraphs of the [Linux maintainer guide](https://docs.brew.sh/Homebrew-linuxbrew-core-Maintainer-Guide#preparation).

Reply:
Added!